### PR TITLE
feat: store only user ID in session

### LINKED ISSUE #212: Should only store `userId` in sessions instead of the entire user object
As per title, should we have a single source of truth for the state of the user?

It currently does not matter much since we have no real features to update the user object once it is created (no email changes, no password changes, etc), and thus we can use `req.session.user` without any thought.

However, with the ability to update the user document now (emergency contact number feature #13), the source of truth may diverge.

Thus, suggest just storing the userId in session and retrieving them whenever needed.

### PR DESCRIPTION
**Pull request by jia1**
Closes #212

### CODE REVIEW

**Review by karrui — APPROVED** [review]
lgtm

**Review by karrui — CHANGES_REQUESTED** [review]
actually, in `src/app/loaders/express/logging.ts`, we have a getter for session.user._id for logging and that may break if we don't update that too

File: src/types/vendor/express.d.ts [hunk 1]
```diff
@@ -7,11 +7,11 @@ declare global {
     }
 
     export interface Session {
-      user?: IUserSchema
+      user?: IUserSchema['_id']
```
Comment (karrui) [inline src/types/vendor/express.d.ts:10]: this should look like 

```ts
user?: {
  _id: IUserSchema['_id']
}
```

File: src/types/vendor/express.d.ts [hunk 2]
```diff
@@ -7,11 +7,11 @@ declare global {
...
     }
 
     export interface AuthedSession extends Session {
-      user: IUserSchema
+      user: IUserSchema['_id']
```
Comment (karrui) [inline src/types/vendor/express.d.ts:14]: Same here

**Review by karrui — CHANGES_REQUESTED** [review]

**Review by karrui — COMMENTED** [review]
should be ok now, just need to fix the build error (remove the user.email call (since session doesn't contain the full user anymore) and replace it with _id i guess))

also search the codebase for instances of `TODO(#212)` and remove them! :D

**Comment by jia1** [comment]
@karrui Do you mean the line containing: `message: Successfully logged in user ${user._id},`? Reference to `.email` was removed but pipeline still fails. [Travis build](https://travis-ci.com/github/opengovsg/FormSG/builds/228306260) gives me an empty screen.

**Comment by karrui** [comment]
> @karrui Do you mean the line containing: `message: Successfully logged in user ${user._id},`? Reference to `.email` was removed but pipeline still fails. [Travis build](https://travis-ci.com/github/opengovsg/FormSG/builds/228306260) gives me an empty screen.

I've fixed it, can check out my commit for more info!

**Review by karrui — APPROVED** [review]
lgtm

**Pull request merged**